File: PokeIn/PokeIn_Free_v2.032/PokeIn_Free_v2.032/PokeInSample/Dummy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading;
using System.Web;
using PokeIn;
using PokeIn.Comet;

//Important! See Global.asax file and read the comments
namespace PokeInSample
{
    public class Dummy:IDisposable
    {
        string _clientId;  

        static Dummy()
        {
            new Thread(delegate()
            {
               while (true)
               {
                   string jsonMethod = JSON.Method("UpdateTime", DateTime.Now);
                   CometWorker.Groups.Send("TimeChannel", jsonMethod);
                   Thread.Sleep(1000);
               }
            }).Start();
        }

        public Dummy(string clientId)
        {
            _clientId = clientId; 
        }

        public void Dispose()
        { 
            // do something on exit
        }

        public void TestString(string str)
        {
            str = JSON.Tidy(str);//to make sure that the string is ready for proper transfer
            CometWorker.SendToClient(_clientId, "UpdateString('" + str + "');");
        } 

        public void GetServerTime()
        {
            string jsonMethod = JSON.Method("UpdateTime", DateTime.Now);
            CometWorker.SendToClient(_clientId, jsonMethod);
        } 

        public void SubscribeToTimeChannel()
        {
            CometWorker.Groups.PinClientID(_clientId, "TimeChannel");
        }

        public void LeaveChannel()
        {
            CometWorker.Groups.UnpinClient(_clientId);
        }
    }
}

File: PokeIn/PokeIn_Free_v2.032/PokeIn_Free_v2.032/VB.NET ServerTimeSample/ServerTimeSample/Dummy.vb
Imports PokeIn
Imports PokeIn.Comet
Imports System.Threading

Public Class Dummy
    Implements IDisposable
    ' Methods
    Shared Sub New()
        Dim ts As ThreadStart = New ThreadStart(AddressOf AppTime)
        Dim th As Thread = New Thread(ts)
        th.Start()
    End Sub

    Shared Function AppTime()
        Do While True
            Dim jsonMethod As String = JSON.Method("UpdateTime", New Object() {DateTime.Now})
            CometWorker.Groups.Send("TimeChannel", jsonMethod)
            Thread.Sleep(&H3E8)
        Loop
        Return vbNull
    End Function

    Public Sub New(ByVal clientId As String)
        Me._clientId = clientId
    End Sub 

    Public Sub GetServerTime()
        Dim jsonMethod As String = JSON.Method("UpdateTime", New Object() {DateTime.Now})
        CometWorker.SendToClient(Me._clientId, jsonMethod)
    End Sub

    Public Sub SubscribeToTimeChannel()
        CometWorker.Groups.PinClientID(Me._clientId, "TimeChannel")
    End Sub

    Public Sub TestString(ByVal str As String)
        str = JSON.Tidy(str)
        CometWorker.SendToClient(Me._clientId, ("UpdateString('" & str & "');"))
    End Sub


    ' Fields
    Private _clientId As String

    Public Sub IDisposable_Dispose() Implements IDisposable.Dispose

    End Sub
End Class
